File: project/donghyeon-onboarding/src/main/java/me/dhlee/donghyeononboarding/service/SurveyService.java
package me.dhlee.donghyeononboarding.service;

import java.util.List;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.util.ObjectUtils;

import lombok.RequiredArgsConstructor;
import me.dhlee.donghyeononboarding.domain.Survey;
import me.dhlee.donghyeononboarding.domain.SurveyItem;
import me.dhlee.donghyeononboarding.domain.SurveyItemOption;
import me.dhlee.donghyeononboarding.dto.request.SurveyCreateRequest;
import me.dhlee.donghyeononboarding.dto.request.SurveyItemCreateRequest;
import me.dhlee.donghyeononboarding.dto.request.SurveyItemOptionCreateRequest;
import me.dhlee.donghyeononboarding.dto.response.SurveyCreateResponse;
import me.dhlee.donghyeononboarding.exception.AppException;
import me.dhlee.donghyeononboarding.exception.ErrorCode;
import me.dhlee.donghyeononboarding.mapper.SurveyMapper;
import me.dhlee.donghyeononboarding.repository.SurveyRepository;

@RequiredArgsConstructor
@Service
public class SurveyService {

    private final SurveyRepository surveyRepository;
    private final SurveyMapper surveyMapper;

    @Transactional
    public SurveyCreateResponse createSurvey(SurveyCreateRequest request) {
        Survey survey = surveyMapper.toEntity(request);
        return SurveyCreateResponse.from(surveyRepository.save(survey));
    }
}

File: project/seonghun-onboarding/src/main/kotlin/com/chosseang/seonghunonboarding/service/SurveyService.kt
package com.chosseang.seonghunonboarding.service

import com.chosseang.seonghunonboarding.dto.SurveyCreateRequest
import com.chosseang.seonghunonboarding.dto.SurveyCreateResponse
import com.chosseang.seonghunonboarding.entity.Survey
import com.chosseang.seonghunonboarding.repository.SurveyRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class SurveyService(val repository: SurveyRepository) {

    @Transactional
    fun createSurvey(surveyCreateRequest: SurveyCreateRequest): SurveyCreateResponse {
        // 각 아이템의 survey 참조 설정 (양방향 관계 유지)
        surveyCreateRequest.items.forEach { item ->
            item.survey = Survey(
                name = surveyCreateRequest.name,
                description = surveyCreateRequest.description,
                items = surveyCreateRequest.items.toMutableList()
            )
        }

        return repository.save(Survey(
            name = surveyCreateRequest.name,
            description = surveyCreateRequest.description,
            items = surveyCreateRequest.items.toMutableList()
        )).let {
            SurveyCreateResponse(
                name = it.name,
                description = it.description,
                items = it.items.toMutableList()
            )
        }
    }

    fun searchSurvey(): List<Survey> {
        return repository.findAll()
    }
}
